trading_calendars/exchange_calendar_xidx.py

```diff
@@ -0,0 +1,447 @@
+    - Labor Day (May 1)
+    - Ascension Day Of Jesus Christ (39 days after Easter, always Thursday)
+    - Pancasila Day (Jun 1)
+    - Independence Day (Aug 17)
+    - Christmas Day (Dec 25)
+    - New Year's Eve (called "Trading Holiday", Dec 31)
+    - Chinese New Year (from Gregorian year 2002 onward)
+    - Islamic New Year
+    - Eid al-Fitr (Festival of Breaking the Fast)
+    - Eid al-Adha (Festival of the Sacrifice)
+    - Isra Mikraj of the Prophet Muhammad
+    - Birth of the Prophet Muhammad
+    - Vesak Day
+    - Hindu Saka New Year (also called Nyepi, or Balinese Day of Silence)
+
+    Election holidays are also observed, as well as additional "common
+    leave" days around many holidays.
+    """
+    name = 'XIDX'
+
+    tz = timezone('Asia/Jakarta')
+
+    open_times = (
+        (None, time(9, 1)),
+    )
+
+    close_times = (
+        (None, time(15, 50)),
+    )
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
+            new_years_day(),
+            GoodFriday,
+            Holiday(
+                "Labor Day",
+                month=5,
+                day=1,
+                start_date='2014-05-01',
+            ),
+            # XXX: The Indonesia Stock Exchange was open on Ascension
+            # Day 2003, but closed the next day. We are not sure why.
+            # (The 2003-05-30 closure is listed in adhoc_holidays.)
+            ascension_day(end_date='2003'),
+            ascension_day(start_date='2004'),
+            Holiday(
+                "Pancasila Day",
+                month=6,
+                day=1,
+                start_date='2017-06-01',
+            ),
+            Holiday(
+                "Independence Day",
+                month=8,
+                day=17,
+            ),
+            # Christmas Eve is not an official Indonesian holiday, but
+            # December 24th and 26th are frequently observed as common
+            # leave. These dates are listed in adhoc_holidays.
+            christmas(),
+            new_years_eve(),
+        ])
+
+    chinese_new_year = chinese_lunar_new_year_dates[
+        # The Indonesia Stock Exchange did not close for Chinese New
+        # Year in 1998, 1999, or 2001. (It fell on a Saturday in 2000.)
+        chinese_lunar_new_year_dates.year >= 2002
+    ]
+
+    common_leave = [
+        # Indonesia sometimes observes additional "common leave" days
+        # around the usual observed dates of holidays.
+
+        # Common leave for New Year's Day.
+        '2002-12-30',
+        '2005-12-30',
+        '2009-01-02',
+        '2017-01-02',
+
+        # Common leave for Chinese New Year.
+        '2008-02-08',
+
+        # Common leave for Ascension Day.
+        '2003-05-30',
+        '2006-05-26',
+        '2007-05-18',
+
+        # Common leave for Independence Day.
+        '2003-08-18',
+        '2006-08-18',
+        '2008-08-18',
+
+        # Common leave for Christmas.
+        '2002-12-24',
```

Comment:
So there was no predictable pattern for when these extra Christmas days were observed?

Reply:
Not that I've been able to find, either from trading data or information online. Note that 2006, 2008, and 2015 had no additional leave.